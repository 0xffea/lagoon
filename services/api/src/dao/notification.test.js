const { Sql } = require('./notification');

describe('Sql', () => {
  describe('selectNotificationsByTypeByProjectId', () => {
    it('should create statement for notification_slack table', () => {
      const cred = {
        role: 'user',
        permissions: { customers: [], projects: [] },
      };

      const input = {
        type: 'slack',
        pid: 3,
      };
      const ret = Sql.selectNotificationsByTypeByProjectId(cred, input);
      expect(ret).toMatchSnapshot();
    });
  });

  describe('selectUnassignedNotificationsByType', () => {
    it('should create a proper query', () => {
      const cred = {
        role: 'user',
        permissions: {
          customers: ['1', '2'],
          projects: [],
        },
      };
      const ret = Sql.selectUnassignedNotificationsByType(cred, 'rocketchat');
      expect(ret).toMatchSnapshot();
    });
  });

  describe('updateNotificationSlack', () => {
    it('should create a proper query', () => {
      const cred = {};
      const input = {
        name: 'n1',
        patch: {
          name: 'test',
          webhook: 'new-webhook',
        },
      };

      const ret = Sql.updateNotificationSlack(cred, input);
      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectCustomer', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectNotificationSlackByName('n1');
      expect(ret).toMatchSnapshot();
    });
  });
});
