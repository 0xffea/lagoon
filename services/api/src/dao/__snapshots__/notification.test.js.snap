// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql selectCustomer should create a proper query 1`] = `"select * from \`notification_slack\` where \`name\` = 'n1'"`;

exports[`Sql selectNotificationsByTypeByProjectId should create statement for notification_slack table 1`] = `"select \`nt\`.*, \`pn\`.\`type\` from \`project_notification\` as \`pn\` JOIN notification_slack AS nt ON pn.nid = nt.id AND pn.type = 'slack' where \`pn\`.\`pid\` = 3"`;

exports[`Sql selectUnassignedNotificationsByType should create a proper query 1`] = `"select \`nt\`.*, 'rocketchat' as type from \`notification_rocketchat\` as \`nt\` left join project_notification AS pn ON pn.nid = nt.id AND pn.type = 'rocketchat' where pn.nid IS NULL and pn.pid IS NULL"`;

exports[`Sql updateNotificationSlack should create a proper query 1`] = `"update \`notification_slack\` set \`name\` = 'test', \`webhook\` = 'new-webhook' where \`name\` = 'n1'"`;
